File: 0442.Find All Duplicates in an Array/trick.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        if(nums.size() == 0)
            return {};
        int n = nums.size();
        vector<int> ret;
        for(int i = 0; i < n; ++i){
            int cur = abs(nums[i]);
            if(nums[cur - 1] < 0){
                // already existing
                ret.push_back(cur);
            }else{
                // marked as existing
                nums[cur - 1] *= -1;
            }
        }
        return ret;
    }
};

File: 0991. Broken Calculator/trick.java
class Solution {
    public int brokenCalc(int startValue, int target) {
        int ret = 0;

        while(target > startValue){
            if(target == startValue)
                return ret;
            if(target % 2 == 1){
                target = target + 1;
            }else{
                target = target / 2;
            }
            ret++;
        }
        return ret + startValue - target;
    }
}